nautilus_trader/data/engine.pyx

```diff
@@ -1583,24 +1566,17 @@ cdef class DataEngine(Component):
         # We assume each symbol is only in one catalog
         for catalog in self._catalogs.values():
             if isinstance(request, RequestInstruments):
-                # For instruments requests, we omit the start parameter to ensure we find
```

Comment:
As above, there are certain cases where handling a range query in this way is logically correct. However, this shouldn't be the default (it's correct to remove this comment to align with standard/previous behavior).